l10n_ca_hr_payroll/l10n_ca_hr_payroll.py

```diff
@@ -1,307 +1,272 @@
+                      based on applicable amounts reported on the provincial 
+                      or territorial Form TD1''')
+    td1 = fields.Float(string='Personal Tax Credits Return (TD1)', 
+                       digits=(16, 2), required=True, default=11038.00, 
+                       help='''\Personal Tax Credits Return''')
+    eeins = fields.Float(string='Insurance - Employee Contribution (EeINS)', 
+                         digits=(16, 2), required=True, default=0.00)
+    erins = fields.Float(string='Insurance - Employer Contribution (ErINS)', 
+                         digits=(16, 2), required=True, default=0.00)
 
-    _defaults = {
-        'td1f': 'code1',
-        'td1p': 'codeA',
-        'td1': 11038.00,
-        'eeins': 0.00,
-        'erins': 0.00,
-    }
 
-
-class hr_contract(orm.Model):
+class HrContract(models.Model):
     _inherit = 'hr.contract'
 
-    def _get_pays_per_year(self, cr, uid, ids, names, arg, context=None):
-        """
-        @param ids: ID of contract
-        @return: The number of pays per year
-        """
-        res = {}
-        # FIXME: Should likely pull these values from somewhere else, depending on whether a 52 or 53 year week is used
+    @api.multi
+    @api.depends('pays_per_year')
+    def _get_pays_per_year(self):
+        last_date = str(datetime.now().year) + '-12-31'
+        weekly = datetime.strptime(last_date, '%Y-%m-%d').strftime("%W")
         schedule_pay = {
-            'weekly': 52,
-            'bi-weekly': 26,
+            'weekly': weekly,
+            'bi-weekly': int(weekly) / 2,
             'monthly': 12,
             'bi-monthly': 6,
             'quarterly': 4,
             'semi-annually': 2,
             'annually': 1,
         }
-        for contract in self.browse(cr, uid, ids, context):
-            if contract.schedule_pay and schedule_pay.get(contract.schedule_pay, False):
-                res[contract.id] = schedule_pay[contract.schedule_pay]
-
-        return res
-
-    _columns = {
-        'pays_per_year': fields.function(
-            _get_pays_per_year, method=True, string='Pays Per Year',
-            type='float', readonly=True,
-        ),
-        'weeks_of_vacation': fields.integer('Number of weeks of vacation', required=True),
-    }
-
-    _defaults = {
-        'weeks_of_vacation': 2,
-    }
+        for contract in self:
+            my_schedule = contract.schedule_pay or False
+            if my_schedule and schedule_pay.get(my_schedule):
+                contract.pays_per_year = schedule_pay[my_schedule]
+                
+    # simple fields
+    pays_per_year = fields.Integer(string='Pays Per Year', readonly=True,
+                                 compute='_get_pays_per_year',
+                                 help='The number of pays per year')
+    weeks_of_vacation = fields.Integer(string='Number of weeks of vacation',
+                                       required=True, default=2)
+    
+#END
```

Comment:
@rigo1985 You need to finish the file with a `\n`. Also, I'm not very found of the `#END` comment at the end (it's a meaningless information) but it's just my personnal taste so it's not blocking.